### PR TITLE
Improve Emmy CLI

### PR DESCRIPTION
**Pull request by mancabizjak**
This PR introduces an improved CLI. All the CLI-related functionality is moved to the new `cli` package. Main function in `emmy.go` is greatly simplified, as it now just instantiates the Emmy CLI app. 

The CLI has the same commands as before, but some major changes are in client and example commands. Protocols are now run as client subcommands. All the CLI flags are defined in a file `flags.go`.  Logic in example CLI command is improved. Finally, we introduce protocol-level flags that were fixed until now, as well as confiruration of server's port and endpoint via CLI.

I also updated the README to include updated CLI instructions.

### CODE REVIEW

File: cli/flags.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+var certFlag = cli.StringFlag{
+	Name:  "cert",
+	Value: filepath.Join(config.LoadTestdataDir(), "server.pem"),
+	Usage: "`PATH` to servers certificate file",
+}
+
+// caCertFlag keeps the path to CA's certificate in PEM format
+// (for establishing a secure channel with the server).
+var caCertFlag = cli.StringFlag{
+	Name:  "caCert",
```
[the remark this answers is not in the document]
Comment (mancabizjak) [reply cli/flags.go]: I was confused why all the `emmy client` subcommands were suddenly panicking (nil pointer dereference) when I realised I forgot to update the flag names `--caCert` and `--pubKey` to `--cacert` and `--pubkey` :woman_facepalming:

File: README.md [hunk 2]
```diff
@@ -4,135 +4,180 @@ Emmy is a library that offers a **crypto-backend** with primitives for implement
...
-3. **Whether to run clients concurrently or not**: flag *--concurrent*. Include this flag if you want to run the specified number of clients consurrently. The absence of this flag means that clients will be run sequentially.
-4. **Logging level**: flag *--loglevel* (shorthand *-l*), which must be one of `debug|info|notice|error|critical`. Defaults to `ìnfo`.
+1. **How many clients to start**: flag *--nclients* (shorthand *-n*), defaults to 1.
+2. **Whether to run clients concurrently or not**: flag *--concurrent*. Include this flag if you want to run the specified number of clients consurrently. The absence of this flag means that clients will be run sequentially.
+3. **Logging level**: flag *--loglevel* (shorthand *-l*), which must be one of `debug|info|notice|error|critical`. Defaults to `ìnfo`.
+4. **URI of the emmy server**: flag *--server*, defaults to *localhost:7007*.
+5. **CA certificate**: flag *--caCert*, points to a path to a certificate of the CA that issued emmy server's certificate, in PEM format. This will be used to secure communication channel with the server. Please refer to [explanation of TLS support in Emmy](#tls-support) for explanation.
```
Comment (miha-plesko) [inline README.md]: `caCert` -> `cacert`

File: README.md [hunk 3]
```diff
@@ -146,31 +191,8 @@ To control keys and certificates used for TLS, emmy CLI programs use several fla
 * `--cert` which expects the path to server's certificate in PEM format, 
 * `--key` which expects the path to server's private key file.
 
-The same flags can be used with `emmy example` program. If they are omitted, files in `test/testdata` directory are used.
-
 On the other hand, we can provide `emmy client` with the `--caCert` flag, which expects the certificate of the CA that issued emmy server's certificate (in PEM format). Again, if this flag is omitted, the certificate in `test/testdata` directory is used.
```
Comment (miha-plesko) [inline README.md]: `--caCert` -> `--cacert`

Comment (mancabizjak) [reply README.md]: Sorry, these flags, again... :smile: You have a good eye :+1:

**Comment by mancabizjak** [comment]
Thanks @miha-plesko for the review and all your suggestions! The final changes are now implemented.

**Pull request merged**